script.js

```diff
@@ -0,0 +1,78 @@
+const previousImage = document.getElementById('left-btn')
+const nextImage = document.getElementById('right-btn')
+const carouselImages = document.querySelectorAll('.carousel__image')
+const carouselIndicatorButtons = document.querySelectorAll(
+	'.carousel__indicator'
+)
+
+let currentImage = 1
+const totalImages = carouselImages.length
+
+function removeActiveNavButtonClass() {
+	carouselIndicatorButtons.forEach((button) => {
+		button.classList.remove('carousel__indicator--active')
+	})
+}
+
+function removeActiveImageClass() {
+	carouselImages.forEach((image) => {
+		image.classList.remove('carousel__image--active')
+	})
+}
+
+function addActiveImageClass(item) {
+	item.classList.add('carousel__image--active')
+}
+
+function addActiveNavButtonClass(index) {
+	removeActiveNavButtonClass()
+	document.getElementById(index).classList.add('carousel__indicator--active')
+}
+
+previousImage.addEventListener('click', () => {
+	removeActiveImageClass()
+
+	if (currentImage - 1 < 0) {
+		addActiveImageClass(carouselImages[carouselImages.length - 1])
+
+		currentImage = carouselImages.length
+
+		addActiveNavButtonClass(currentImage)
+		currentImage--
+	} else {
+		addActiveImageClass(carouselImages[currentImage - 1])
+
+		addActiveNavButtonClass(currentImage)
+		currentImage--
+	}
+})
+
+nextImage.addEventListener('click', () => {
+	removeActiveImageClass()
+
+	if (currentImage + 1 >= carouselImages.length + 1) {
+		addActiveImageClass(carouselImages[carouselImages.length - 1])
+
+		currentImage = 1
+		addActiveNavButtonClass(currentImage)
+	} else {
+		addActiveImageClass(carouselImages[currentImage - 1])
+
+		currentImage++
+		addActiveNavButtonClass(currentImage)
+	}
+})
+
+carouselIndicatorButtons.forEach((button) => {
+	button.addEventListener('click', () => {
+		const imageSelected = carouselImages[button.id - 1]
```

Comment:
I see some playing around with indices. Consider simplifying a bit ;) Don't take long ;) Up to 30 mins.